### PR TITLE
Fix samples to run on iOS arm64 simulators

### PR DESCRIPTION
**Pull request by jpmartin**
## Description
Update sample projects to run on x64, arm64 and arm64Simulators.

## Motivation and Context
The configuration is based on the [official documentation](https://kotlinlang.org/docs/native-cocoapods.html):
- Add missing presets in `build.gradle.kts`.
- Remove `copyFramework` task
- Add `syncFramework` task in podspec
- Add a reference to `translation.en.json` in Xcode project instead of copying it into the framework.
- Run `syncFramework` in Podfile to prevent compile errors after a clean install.

## How Has This Been Tested?
The samples apps now run on the simulator and device.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec [hunk 1]
```diff
@@ -4,47 +4,39 @@ Pod::Spec.new do |spec|
...
-  cd "$SRCROOT/../../../.."
-
-  ./gradlew :trikot-viewmodels-declarative:sample:common:copyFramework \
-    -Pconfiguration.build.dir="build/bin/ios" \
-    -Pkotlin.build.type="$KOTLIN_BUILD_TYPE" \
-    -Pkotlin.target="$KOTLIN_TARGET"
-else
-  echo "Skipping build of common framework in preview mode"
-fi
+                if [ "YES" = "$COCOAPODS_SKIP_KOTLIN_BUILD" ]; then
```
Comment (antoinelamy) [inline trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:29]: Not exactly sure when `COCOAPODS_SKIP_KOTLIN_BUILD` is defined to `YES` but we are probably gonna need to continue to support `ENABLE_PREVIEWS` to make sure SwiftUI previews don't break. Unless CocoaPods proxy that value through their own environment variable but I can't find the documentation on this flag.

File: trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec [hunk 2]
```diff
@@ -4,47 +4,39 @@ Pod::Spec.new do |spec|
     spec.homepage                 = 'www.mirego.com'
     spec.source                   = { :git => "Not Published", :tag => "Cocoapods/#{spec.name}/#{spec.version}" }
     spec.authors                  = ''
-    spec.license                  = 'No license'
-    spec.summary                  = 'Awesome library'
+    spec.license                  = ''
```
Comment (antoinelamy) [inline trikot-viewmodels-declarative/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:7]: spec.license      = { :type => 'BSD' }

File: trikot-viewmodels/sample/common/TRIKOT_FRAMEWORK_NAME.podspec [hunk 3]
```diff
@@ -1,48 +1,42 @@
...
-    spec.vendored_frameworks      = "build/bin/ios/TRIKOT_FRAMEWORK_NAME.framework"
-    spec.libraries                = "c++", "System"
+    spec.vendored_frameworks      = "build/cocoapods/framework/TRIKOT_FRAMEWORK_NAME.framework"
+    spec.libraries                = "c++"
     spec.module_name              = "#{spec.name}_umbrella"
 
+                
+
+                
+
```
Comment (antoinelamy) [inline trikot-viewmodels/sample/common/TRIKOT_FRAMEWORK_NAME.podspec:17]: Extra lines

**Comment by jpmartin** [comment]
Using the CocoaPods plugin requires further investigation. For now we will use https://github.com/mirego/trikot/pull/29.

**Pull request closed without merging**